som-interpreter-bc/src/interpreter.rs

```diff
@@ -266,15 +329,76 @@ impl Interpreter {
                     };
                     let signature = universe.lookup_symbol(symbol);
                     let nb_params = nb_params(signature);
+                    let holder = frame.borrow().get_method_holder();
+                    let method = {
+                        let super_class = holder.borrow().super_class()?;
+                        match frame.borrow().kind() {
+                            FrameKind::Block { block } => {
+                                let mut inline_cache_receiver =
```

Comment:
We are here at the super send, right?
At this point, we should not need the receiver.
The super send should be "lexically bound". So, it's the super class of the class where the method with the `super` is in, which is subtly different from being receiver-related.

Reply:
This is indeed the code for the super sends.  

The method lookup is done using the super class of the current method holder ("method holder" being the term I use to mean the class that defines the method), but the receiver of the message (the `self_value` later defined in the code) is different from that.  
So, the code behaves correctly but the naming is indeed wrong and/or misleading.  

This will be addressed if this particular inline cache implementation is the one that gets to stay.

Reply:
After my initial performance measurements, it seems that the technique using a vector of `(class, method)` tuples is faster than using two separate vectors.  
In the new implementation, the variable is simply called `inline_cache`, which should resolve the possible confusion that was previously possible.